packages/firestore/src/local/local_store.ts

```diff
@@ -869,26 +868,11 @@ export class LocalStore {
     const mode = keepPersistedTargetData ? 'readwrite' : 'readwrite-primary';
     return this.persistence
       .runTransaction('Release target', mode, txn => {
-        // References for documents sent via Watch are automatically removed
-        // when we delete a target's data from the reference delegate.
-        // Since this does not remove references for locally mutated documents,
-        // we have to remove the target associations for these documents
-        // manually.
-        // This operation needs to be run inside the transaction since EagerGC
-        // uses the local view references during the transaction's commit.
-        // Fortunately, the operation is safe to be re-run in case the
-        // transaction fails since there are no side effects if the target has
-        // already been removed.
-        const removed = this.localViewReferences.removeReferencesForId(
-          targetId
-        );
-
         if (!keepPersistedTargetData) {
-          return PersistencePromise.forEach(removed, (key: DocumentKey) =>
-            this.persistence.referenceDelegate.removeReference(txn, key)
-          ).next(() => {
-            this.persistence.referenceDelegate.removeTarget(txn, targetData!);
-          });
+          return this.persistence.referenceDelegate.removeTarget(
```

Comment:
The reference delegate now knows internally what documents are orphaned in the view.